conans/client/rest/rest_client_v1.py

```diff
@@ -220,7 +220,7 @@ def upload_recipe(self, conan_reference, the_files, retry, retry_wait, ignore_de
             deleted.remove(ignore_deleted_file)
 
         if not new and not deleted and modified in (["conanmanifest.txt"], []):
-            return False, conan_reference
+            return False, conan_reference.copy_without_revision()
```

Comment:
Not sure why need a copy_without_revision, as it is V1, in theory the references should already come without revision?

Reply:
Because that "conan_reference" comes from the user input (keyboard), in the protocol is discarded, so the server is not using the revision, but in that point is returning the same input reference. Maybe is it easier to raise if the reference contains a revision in the apiv1. I will change it and see what happen.

Reply:
Maybe at the second line:

```
# In case the recipe comes from a revisioned remote we clean the revision because current remote doesn't support it
conan_reference = conan_reference.copy_without_revision()
```

Reply:
Yes, maybe cleaning it after getting from the registry is what makes sense

Reply:
Yes, it is cleaned after the upload (to store that it is not revisioned). But the line above is needed at the beginning.